packages/devtools_app/lib/src/analytics/remote_provider.dart

```diff
@@ -4,74 +4,65 @@
 
 import 'dart:async';
 
+import 'package:flutter/foundation.dart';
+
 import '../config_specific/server/server.dart' as server;
 import 'analytics.dart' as analytics;
 import 'provider.dart';
 
 class _RemoteAnalyticsProvider implements AnalyticsProvider {
   _RemoteAnalyticsProvider(
-    this._isEnabled,
-    this._isFirstRun,
-    this._isGtagsEnabled,
-  );
+    bool enabled,
+    bool firstRun,
+  )   : _analyticsEnabled = ValueNotifier<bool>(enabled),
+        _shouldPrompt = ValueNotifier<bool>(firstRun && !enabled);
 
   @override
-  bool get isEnabled => _isEnabled;
-  final bool _isEnabled;
+  ValueListenable<bool> get analyticsEnabled => _analyticsEnabled;
+  final ValueNotifier<bool> _analyticsEnabled;
 
   @override
-  bool get shouldPrompt => _isFirstRun;
-  final bool _isFirstRun;
+  ValueListenable<bool> get shouldPrompt => _shouldPrompt;
+  final ValueNotifier<bool> _shouldPrompt;
 
-  @override
-  bool get isGtagsEnabled => _isGtagsEnabled;
-  final bool _isGtagsEnabled;
+  bool analyticsInitialized = false;
 
   @override
-  void setAllowAnalytics() {
-    setUpAnalytics();
-    analytics.setAllowAnalytics();
+  Future<void> enableAnalytics() async {
+    _analyticsEnabled.value = await analytics.enableAnalytics();
   }
 
   @override
-  void setDontAllowAnalytics() {
-    analytics.setDontAllowAnalytics();
+  Future<void> disableAnalytics() async {
+    _analyticsEnabled.value = await analytics.disableAnalytics();
+    _shouldPrompt.value = false;
```

Comment:
Seems like we could set 
_shouldPrompt.value = false; before awaiting the async operation. Defensively could avoid cases where the prompt was shown multiple times.

Reply:
done